### DISCUSSION TITLE
Add type definitions for typescript

### DISCUSSION THREAD

**Discussion by macliems** [vimeo/player.js#679] [General]
angular-cli:

After npm install @vimeo/player, import fails with 'Cannot find module '@vimeo/player'.

I tried:

import Player from '@vimeo/player'
import Player from '@vimeo/player/dist/player'
import * as Player from '@vimeo/player'

**Comment by bdougherty** [comment]
Sounds like this might be an issue with angular-cli not being able to resolve scoped packages. Probably best to file an issue on `angular-cli`.

**Comment by macliems** [comment]
angular-cli wants typings

I installed @typings/vimeo but that didn't work either

**Comment by bdougherty** [comment]
So it sounds like you’re using typescript? Happy to accept a pull request that adds whatever metadata is needed for that.

**Comment by bdougherty** [comment]
Also looks like you can request that they add definitions for this library to https://github.com/DefinitelyTyped/DefinitelyTyped.

**Comment by denisyilmaz** [comment]
Hi @bdougherty, I've created the typings for @vimeo/player following your README.md file. I've created a pull request at DefinitelyTyped >> https://github.com/DefinitelyTyped/DefinitelyTyped/pull/15593

I don't know how to implement it in the API itself, but maybe somebody else could take it from here?

**Comment by denisyilmaz** [comment]
Types are now available via @types 
>> DefinitelyTyped/DefinitelyTyped@a209273

Till Typescript 2.3.1 (which will support resolution of scopped packages) is out you have to install it via `npm install @types/vimeo__player`.

**Comment by philcockfield** [comment]
Would be nice if this were re-opened and the types included directly in the project.  

TypeScript is now arguably as mainstream as we can get, and this would definately provide a level of confidence that Vimeo was "onto it" if the player client libs "just worked" with types!

Love you guys!

**Comment by anjared** [comment]
I'm using Typescript and had installed both the `@vimeo/player` & `@types/vimeo__player` packages.  I was still getting this error and found the solution here:  

The example code in this comment worked for me: https://github.com/vimeo/player.js/issues/81#issuecomment-262087843

**Comment by bdougherty** [comment]
Going to close this out because the original issue was not entirely related to TypeScript. Please join #901 for the TS discussion.
